akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -221,6 +222,15 @@ class MaterializationException(msg: String, cause: Throwable = null) extends Run
 final case class AbruptTerminationException(actor: ActorRef)
   extends RuntimeException(s"Processor actor [$actor] terminated abruptly") with NoStackTrace
 
+/**
+ * Signal that the stage was abruptly terminated, usually seen as a call to `postStop` of the `GraphStageLogic` without
+ * any of the handler callbacks seeing completion, cancellation or failure from up or downstream. This can happen when
```

Comment:
I mean that we should always spell out that cancellations come from downstream and failures come from upstream.

"cancellation from downstream or failure from upstream"